fontbe/src/colr.rs

```diff
@@ -14,21 +26,87 @@ pub fn create_colr_work() -> Box<BeWork> {
     Box::new(ColrWork {})
 }
 
+fn to_colr_paint(glyph_order: &GlyphOrder, palette: &ColorPalettes, ir_paint: &ir::Paint) -> Paint {
+    match ir_paint {
+        ir::Paint::Glyph(paint) => Paint::Glyph(PaintGlyph {
+            paint: to_colr_paint(glyph_order, palette, &paint.paint).into(),
+            glyph_id: glyph_order
+                .glyph_id(&paint.name)
+                .expect("Validated earlier"),
+        }),
+        ir::Paint::Solid(paint) => Paint::Solid(PaintSolid {
+            palette_index: palette.index_of(paint.color).unwrap_or(FOREGROUND_COLOR) as u16,
+            alpha: OPAQUE,
+        }),
+    }
+}
+
 impl Work<Context, AnyWorkId, Error> for ColrWork {
     fn id(&self) -> AnyWorkId {
         WorkId::Colr.into()
     }
 
     fn read_access(&self) -> Access<AnyWorkId> {
-        Access::Variant(FeWorkId::ColorPalettes.into())
+        AccessBuilder::new()
+            .variant(FeWorkId::ColorPalettes)
+            .variant(WorkId::ALL_GLYF_FRAGMENTS)
+            .specific_instance(FeWorkId::GlyphOrder)
+            .build()
     }
 
     /// Generate [COLR](https://learn.microsoft.com/en-us/typography/opentype/spec/colr)
     fn exec(&self, context: &Context) -> Result<(), Error> {
-        // Temporary: if there is a palette generate an empty COLR, just so we can easily see a ttx_diff
-        if context.ir.colors.try_get().is_some() {
-            context.colr.set(Default::default());
+        let Some(paint_graph) = context.ir.paint_graph.try_get() else {
+            return Ok(());
+        };
+        let palette = context.ir.colors.try_get().unwrap_or_default();
+        let glyph_order = context.ir.glyph_order.get();
+
+        let mut colr = Colr::new(0, None, None, 0);
+        colr.base_glyph_list = BaseGlyphList::new(
+            paint_graph.base_glyphs.len() as u32,
+            paint_graph
+                .base_glyphs
+                .iter()
+                .map(|(glyph_name, paint)| {
+                    BaseGlyphPaint::new(
+                        glyph_order.glyph_id(glyph_name).unwrap(),
+                        to_colr_paint(&glyph_order, &palette, paint),
+                    )
+                })
+                .collect(),
+        )
+        .into();
+
+        let mut clips = Vec::<Clip>::new();
+        for glyph_name in paint_graph.base_glyphs.iter().map(|(g, _)| g) {
+            let next_gid = glyph_order.glyph_id(glyph_name).expect("Validated earlier");
+            let next_glyph = context
+                .glyphs
+                .get(&WorkId::GlyfFragment(glyph_name.clone()).into());
+            let next_clip = next_glyph.data.bbox().unwrap_or_default();
```

Comment:
ufo2ft quantizes clipBoxes to 1/10th of the upem (e.g. 100 units for 1000 UPEM) to increase the chance of them getting reused. Search for "colrClipBoxQuantization" in ufo2ft.
Maybe for later